src/ThriveWorkerUnit.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/**
+ * @title ThriveWorkerUnit
+ * @dev Contract of a work unit representing a task where contributors complete and earn rewards.
+ */
+contract ThriveWorkerUnit {
+    address public immutable moderator;
+    address public immutable rewardToken;
+    uint256 public immutable rewardAmount;
+    uint256 public immutable maxRewards;
+    uint256 public immutable deadline;
+    uint256 public immutable maxCompletionsPerUser;
+    address public immutable assignedAddress;
```

Comment:
Required Badge is missing as well, I would suggest to have bytes32 array as we might want one of multiple badges